### PR TITLE
Fix typos discovered by codespell

### PR DESCRIPTION
**Pull request by cclauss**
@xmatthias as discussed elsewhere.

% `codespell --ignore-words-list=coo,fo,lowd,nonesense,strat,zar --skip="*.svg" --write-changes`

### CODE REVIEW

File: freqtrade/persistence/trade_model.py [hunk 1]
```diff
@@ -1502,7 +1502,7 @@ class Trade(ModelBase, LocalTrade):
     id: Mapped[int] = mapped_column(Integer, primary_key=True)  # type: ignore
 
     orders: Mapped[List[Order]] = relationship(
-        "Order", order_by="Order.id", cascade="all, delete-orphan", lazy="selectin",
+        "Order", order_by="Order.id", cascade="all, delete-orphan", lazy="selection",
```
Comment (cclauss) [inline freqtrade/persistence/trade_model.py]: Careful review, please.

File: tests/freqtradebot/test_freqtradebot.py [hunk 2]
```diff
@@ -4829,7 +4829,7 @@ def refresh_latest_ohlcv_mock(pairlist, **kwargs):
     freqtrade.execute_entry('ETH/USDT', 123, is_short=is_short)
     freqtrade.execute_entry('LTC/USDT', 2.0, is_short=is_short)
     freqtrade.execute_entry('XRP/USDT', 123, is_short=is_short)
-    multipl = 1 if is_short else -1
```
Comment (cclauss) [inline tests/freqtradebot/test_freqtradebot.py:4832]: Careful review, please.

File: freqtrade/persistence/trade_model.py (hunk 1 above)
Comment (xmatthias) [reply freqtrade/persistence/trade_model.py]: Yeah, this one explicitly refers to `selectin` in the sqlalchemy "world" - selection is not correct here, and makes essentially every test fail (at least the ones that touch this file)

**Review by xmatthias — APPROVED** [review]
That's great! :100: 
Thanks for taking the time to do this (and also figure out the proper ignores).

I've taken the liberty to use this PR to add a minimal configuration for codespell - so it's easier to run (consistently) in the future.

I'm also considering adding this to the pre-commit hooks - which would make this a mandatory requirement for new code to be pushed in.

---

What i found quite interesting is that while it does fix spelling mistakes (which is great, don't get me wrong!) - it didn't fix some "simple" grammar (like, repeated words).
(after researching a bit): there seems to be an open issue for that (https://github.com/codespell-project/codespell/issues/3160) - so it's clearly not supported by codespell at this point.

**Pull request merged**